src/solvers/smt2_incremental/convert_expr_to_smt.cpp

```diff
@@ -557,6 +557,13 @@ static smt_termt convert_expr_to_smt(const shift_exprt &shift)
       convert_expr_to_smt(left_shift->op0()),
       convert_expr_to_smt(left_shift->op1()));
   }
+  else if(
+    const auto right_logical_shift = expr_try_dynamic_cast<lshr_exprt>(shift))
+  {
+    return smt_bit_vector_theoryt::logical_shift_right(
+      convert_expr_to_smt(right_logical_shift->op0()),
```

Comment:
Can you separate out the conversion of `op0` and `op1` into temporary variables at the top of the function instead, to avoid duplicating the call of the conversion in each of the if blocks?

Reply:
This has been done :)